osm: add documentation for machine rotation (#1087)
kubermatic/docs: content/kubermatic/master/tutorials_howtos/operating_system_manager/_index.en.md

| Openstack | ✓ | ✓ | ✓ | x |
| VSphere | ✓ | ✓ | ✓ | x |

## Machine rotation

> When there is a change in the Machine Deployment template then the Machine set and in-turn the machines are rotated; nodes are re-created. OSM will take care of re-generating the OperatingSystemConfigs and provisioning secrets when there is a change in the Machine Deployment template. So the newly created machines will pick up the new configs.

Reviewer: ```suggestion
When there is a change in the Machine Deployment template then the Machine set and in-turn the machines are rotated; nodes are re-created. OSM will take care of re-generating the OperatingSystemConfigs and the provisioning secrets. So any newly created machine will pick up the new configs.
```

> Although, if an `OperatingSystemProfile` is updated then it wouldn't result in an automatic rotation of the machines. This is an intentional design decision since an `OperatingSystemProfile` can be associated with multiple Machine Deployments and it is the user's responsibility to rotate the machines when the `OperatingSystemProfile` is updated.

Reviewer: ```suggestion
However, if an `OperatingSystemProfile` is updated, that wouldn't result in an automatic rotation of the machines. This is an intentional design decision since an `OperatingSystemProfile` can be associated with multiple Machine Deployments and it is the user's responsibility to rotate the machines when the `OperatingSystemProfile` is updated.
```

> To overcome this limitation, users need to update the `MachineDeployments` manually. An example for a somewhat benign change that could trigger rotation is to update the `.spec.templates.metadata.annotations` field of a `MachineDeployment`. This would result in the annotation being added to the machines and the machines would be rotated.

Reviewer: `Or simply delete the machines` ?

```yaml
apiVersion: "cluster.k8s.io/v1alpha1"
kind: MachineDeployment
metadata:
  name: aws-machinedeployment
  namespace: kube-system
spec:
  selector:
    matchLabels:
      foo: bar
  template:
    metadata:
      labels:
        foo: bar
       annotations:
        force-rotate: true
```

## Limitations

> - machine-controller is responsible for generating bootstrap configuration. In the future OSM will take care of this.

Reviewer: ```suggestion
- machine-controller is responsible for generating bootstrap configuration. In the future OSM will take care of this as well.
```
